test/installplan.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func WaitForInstallPlan(ctx *Context, namespace string, csvName, olmSource string) (*v1alpha1.InstallPlan, error) {
+	var plan *v1alpha1.InstallPlan
+	if waitErr := wait.PollImmediate(Interval, 15*time.Minute, func() (bool, error) {
+		installPlans, err := ctx.Clients.OLM.OperatorsV1alpha1().InstallPlans(namespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, installPlan := range installPlans.Items {
+			if installsCSVFromSource(installPlan, csvName, olmSource) {
+				plan = &installPlan
+				return true, nil
+			}
+		}
+		return false, nil
+	}); waitErr != nil {
+		return plan, fmt.Errorf("installplan for CSV %s and OLM source %s not found: %w", csvName, olmSource, waitErr)
+	}
+	return plan, nil
+}
+
+func installsCSVFromSource(installPlan v1alpha1.InstallPlan, csvName, olmSource string) bool {
+	if installPlan.Status.BundleLookups[0].CatalogSourceRef.Name != olmSource {
+		return false
+	}
+	for _, name := range installPlan.Spec.ClusterServiceVersionNames {
+		if name == csvName {
+			return true
+		}
+	}
+	return false
+}
+
+func ApproveInstallPlan(ctx *Context, name string) error {
+	plan, err := ctx.Clients.OLM.OperatorsV1alpha1().InstallPlans(OperatorsNamespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	plan.Spec.Approved = true
+	_, err = ctx.Clients.OLM.OperatorsV1alpha1().InstallPlans(OperatorsNamespace).Update(context.Background(), plan, metav1.UpdateOptions{})
```

Comment:
We should use either use Patch, or use RetryOnConflict   (note we hit issues before we switched to "oc patch" for installplan approval , see https://github.com/openshift-knative/serverless-operator/pull/727

Reply:
Right. Previously I was looking at the other places in the repo and what's used there. It's the Update method but there we use it in the "reconciler" loop so it will be reapplied if it fails. Just pushed a fix using the Patch.